api/server/middleware/middleware.go

```diff
@@ -112,6 +114,40 @@ func AcceptHeaderHandler(serverAcceptedTypes []string) Middleware {
 	}
 }
 
+func GzipMiddleware() Middleware {
```

Comment:
Can you rename this to `AcceptEncodingHeaderHandler` because we already have `AcceptHeaderHandler`?